#164796896 user should be able to follow and unfollow eachother

Raymond-Osy opened the pull request:
#### Description
- This is an API endpoint for a user to be able to follow and unfollow another user

#### Type of change
- New feature (non-breaking change that adds functionality)

#### How Has This Been Tested?
- [x] Integration test
- [x] Unit test

#### Checklist:
- [x] Style guide

#### PT
[#164796896](https://www.pivotaltracker.com/story/show/164796896)

#### Snap shots
User following postman test
<img width="802" alt="Screenshot 2019-04-16 at 8 54 12 PM" src="https://user-images.githubusercontent.com/35389434/56240026-4e7bcb80-608a-11e9-872a-7c866a90f30d.png">

User unfollowing postman test
<img width="798" alt="Screenshot 2019-04-16 at 8 54 36 PM" src="https://user-images.githubusercontent.com/35389434/56240062-681d1300-608a-11e9-9e63-7a8b2e0a0d4f.png">

ssanusi commented on src/controllers/follower.js:
```diff
@@ -0,0 +1,110 @@
+          status: 'fail',
+          message: 'Sorry, you can not unfollow yourself'
+        })
+      );
```
@Raymond-Osy you can implement follow and unfollow as a toggle functionality. click to follow. once you click again you unfollow that way you can use findOrCreate you already have for following functionality

micah-akpan replied:
@Raymond-Osy. Nice work.  you can modify the error response to only return  `status` and `data` in the body.

Raymond-Osy replied:
Thanks, @ssanusi for the feedback, I never thought of implementing it that way. I have implemented these changes.

Raymond-Osy replied:
Thanks @micah-akpan for spotting it, I've adjusted it accordingly

ssanusi commented on src/routers/follower.js:
```diff
@@ -0,0 +1,7 @@
+// import { Router } from 'express';
```
@Raymond-Osy remember to remove this code if they are not needed

Raymond-Osy replied:
Sure, thanks

ssanusi commented on src/routers/index.js:
```diff
@@ -58,6 +60,10 @@ router.get('/auth/google/callback', passport.authenticate('google', { failureRed
   .post('/signup', checkBody, createUser);
+router.post('/login', checkFields, login);
+router.post('/followers/:followerId/follow', Authenticator.verifyToken, followUser);
+router.delete('/followers/:followerId/unfollow', Authenticator.verifyToken, unfollowUser);
```
@Raymond-Osy if you implement as toggle functionality you don't need  unfollow endpoint

Raymond-Osy replied:
Yes, thanks for the idea...it makes sense. Implemented.

ssanusi reviewed (commented):
@Raymond-Osy nice work I see you considered most edge cases but you could have one endpoint to implement follow and unfollow

micah-akpan commented:
@Raymond-Osy please Squash the commits. Allowable number of commits on a PR is 1 to 5 commits. Thanks

Raymond-Osy committed:
164796896-story(userFollowing): implement user following and unfollowing API endpoint
- add API endpoint for follow and unfollowing a user
- implement teams feedback on using one endpoint
- write all test cases
[Delivers #164796896]
